<launch>

    <arg name="how_many_robot" value="3"/>
    <param name="explore-lite" value="true"/>
    <arg name="cluster_radius" value="2.7"/>
    <param name="cluster_radius" value="$(arg cluster_radius)"/>
    <arg name="world" value="office_big"/>
    <param name="world" value="$(arg world)"/>

    <!-- world -->
    <include file="$(find test_unknown_rendezvous)/launch/world.launch">
        <arg name="use_gui" value="true"/>
        <arg name="world_name" value="$(arg world)"/>
    </include>

    <!-- robot 1 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="1"/>
        <arg name="pose_x"       value="-37"/>
        <arg name="pose_y"       value="22"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="true"/>
        <arg name="formation"    value="false"/>
        <arg name="exploration"  value="true"/>
        <arg name="blobbing"     value="false"/>
    </include>

    <!-- robot 2 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="2"/>
        <arg name="pose_x"       value="-5"/>
        <arg name="pose_y"       value="23"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="true"/>
        <arg name="formation"    value="false"/>
        <arg name="exploration"  value="true"/>
        <arg name="blobbing"     value="false"/>
    </include>
    
    <!-- robot 3 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="3"/>
        <arg name="pose_x"       value="42"/>
        <arg name="pose_y"       value="27"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="true"/>
        <arg name="formation"    value="false"/>
        <arg name="exploration"  value="true"/>
        <arg name="blobbing"     value="false"/>
    </include>



    <!-- all_positions -->
    <node 
        name   = "pos_aggregator"
        pkg    = "test_unknown_rendezvous"
        type   = "pos_aggregator.py"
        args   = "$(arg how_many_robot)"
        output = "screen"
    />
    
    <!-- cluster -->
    <node 
        name   = "cluster"
        pkg    = "test_unknown_rendezvous"
        type   = "cluster_controller.py"
        args   = "$(arg cluster_radius) $(arg how_many_robot)"
        output = "screen"
    />

    <!-- RVIZ --> 
    <include file="$(find test_unknown_rendezvous)/launch/rviz.launch">
        <arg name="rviz_file" value="test_8"/>
    </include> 

</launch>